expenses/controllers/RequisitionController.py

```diff
@@ -1,19 +1,73 @@
-from expenses.models import Requisition
+from django.db.models import Max
+from itertools import zip_longest
+from expenses.models import Requisition, Project, Vendor, RequisitionItem
 
 
 class RequisitionController:
-    @classmethod
-    def can_view(cls, user):
-        return user.has_perm("expenses.view_requisition")
-
     @classmethod
     def get_requisition(cls, info, year, short_code, project_requisition_id):
-        if cls.can_view(info.context.user):
+        if info.context.user.has_perm("expenses.view_requisition"):
             return Requisition.objects.get(project__year=year,
                                            project__short_code=short_code,
                                            project_requisition_id=project_requisition_id)
 
     @classmethod
     def get_requisitions(cls, info):
-        if cls.can_view(info.context.user):
-            return Requisition.objects.filter(created_by=info.context.user, project__archived=False)
\ No newline at end of file
+        if info.context.user.has_perm("expenses.view_requisition"):
+            return Requisition.objects.filter(created_by=info.context.user, project__archived=False)
+
+    @classmethod
+    def create_requisition(cls, info, data):
+        if info.context.user.has_perm("expenses.add_requisition"):
+            project = Project.objects.get(id=data["project"])
+            id_max = Requisition.objects.filter(project=project).aggregate(Max('project_requisition_id'))
+            new_data = {
+                "project": project,
+                "vendor": Vendor.objects.get(id=data["vendor"]) if "vendor" in data else None,
+                "project_requisition_id": id_max["project_requisition_id__max"] + 1,
```

Comment:
This appears to break when you try to create the first requisition in a project

Steps to reproduce:

1. Create a new project
2. Fill out new requisition form
3. Click either submit button and you get the error `graphql.error.located_error.GraphQLLocatedError: unsupported operand type(s) for +: 'NoneType' and 'int'`
4. Open a new tab and go to Django's admin panel.  Create a requisition for the project
5. Go back to your requisition and submitting will work

Reply:
Fixed